src/uct/ib/ud/base/ud_ep.c

```diff
@@ -685,23 +685,12 @@ static uct_ud_ep_t *uct_ud_ep_create_passive(uct_ud_iface_t *iface, uct_ud_ctl_h
 }
 
 static void uct_ud_ep_rx_ctl_drop_packet(uct_ud_ep_t *ep, uct_ud_neth_t *neth,
+                                         uint16_t exp_flags,
                                          const char *packet_type_str)
 {
-    uct_ud_ctl_hdr_t *ctl = (uct_ud_ctl_hdr_t*)(neth + 1);
-    int creq_received     = (ep->flags & UCT_UD_EP_FLAG_CREQ_RCVD) &&
-                            (ep->rx_creq_count > 1);
-    int crep_received     = ep->flags & UCT_UD_EP_FLAG_CREP_RCVD;
-
     ucs_trace_data("ep %p: drop %s with psn %u, head_sn %u",
-                   ep, packet_type_str, neth->psn, ep->rx.ooo_pkts.head_sn);
-
-    ucs_assertv_always(/* Drop CREQ, if the endpoint received duplicate
-                        * CREQ */
-                       ((ctl->type == UCT_UD_PACKET_CREQ) && creq_received) ||
-                       /* Drop:
-                        * - CREP, if the endpoint received duplicate CREP
-                        * - CREQ, if the endpoint received CREQ after CREP */
-                       crep_received,
+                   ep, packet_type_str, neth->psn, ep->rx.ooo_pkts.head_sn);\
+    ucs_assertv_always(ep->flags & exp_flags, /* At lease one must be set */
```

Comment:
minor: print exp_flags
can force push after adding this

Reply:
done